source/checkpoint_restart.f90

```diff
@@ -165,728 +248,349 @@ subroutine cr_killSwitch(iTurn)
     end if
 
-    if(melens .gt. 0) then
-      write(93,"(a)") "SIXTRACR> CRCHECK reading fort.96 Record 9 ELENS"
-      flush(93)
-      call elens_crcheck(96,lerror)
-      if (lerror) goto 101
-    endif
-
-    !ERIC new extended checkpoint for synuthck
-    if (crsythck) then
-      !ERICVARS
-      ! and make sure we can read the extended vars before leaving fort.96
-      ! We will re-read them in crstart to be sure they are correct
-      write(93,"(a,i0)") "SIXTRACR> CRCHECK verifying Record 10 extended vars fort.96, crnapxo=",crnapxo
-      flush(93)
-      write(93,"(a)") "SIXTRACR> CRCHECK verifying extended vars fort.96"
-      flush(93)
-      read(96,end=101,err=101,iostat=ierro)                 &
+    if(melens > 0) then
+      write(crlog,"(a)") "CR_CHECK>  * ELENS variables"
+      flush(crlog)
+      call elens_crcheck(cr_pntUnit(nPoint),rErr)
+      if(rErr) cycle
+    end if
+
+    ! New extended checkpoint for synuthck (ERIC)
+    if(cr_sythck) then
+      ! and make sure we can read the extended vars before leaving fort.95
+      ! We will re-read them in crstart to be sure they are restored correctly
+      write(crlog,"(a)") "CR_CHECK>  * THICK EXTENDED arrays"
+      flush(crlog)
+      read(cr_pntUnit(nPoint),iostat=ioStat) &
         ((((al(k,m,j,l),l=1,il),j=1,crnapxo),m=1,2),k=1,6), &
         ((((as(k,m,j,l),l=1,il),j=1,crnapxo),m=1,2),k=1,6), &
         (dpd(j),j=1,crnapxo),(dpsq(j),j=1,crnapxo),(fokqv(j),j=1,crnapxo)
-      backspace (96,iostat=ierro)
-      write(93,"(a)") "SIXTRACR> CRCHECK read fort.96 EXTENDED OK"
-      flush(93)
-      write(93,"(a)") "SIXTRACR> CRCHECK Leaving fort.96 for CRSTART EXTENDED"
-      flush(93)
+      backspace(cr_pntUnit(nPoint),iostat=ioStat)
+      if(ioStat /= 0) cycle
+      write(crlog,"(a)") "CR_CHECK> Read "//cr_pntFile(nPoint)//" EXTENDED OK"
+      write(crlog,"(a)") "CR_CHECK> Leaving "//cr_pntFile(1)//" for CRSTART EXTENDED"
+      flush(crlog)
     end if
-    read96=.true.
-    goto 103
-  end if
-101 continue
-  if (.not.read96) then
-    write(93,"(a)") "SIXTRACR> CRCHECK ERROR Could not read checkpoint file.96"
-    flush(93)
-  end if
-103 continue
 
-  ! If we have successfully read either fort.95 or fort.96
-  ! we need to handle lost particles and ntwin .ne. 2
-  ! Otherwise we just continue with checkpointing as requested
-  if(read95 .or. read96) then
-    write(93,"(2(a,l1),7(a,i0))") "SIXTRACR> CRCHECK read95=",read95,", read96=",read96,&
-      ", crnapxo=",crnapxo,", crbinrec=",crbinrec,", napx=",napx,", sixrecs=",sixrecs,  &
-      ", crsixrecs=",crsixrecs,", crbnlrec=",crbnlrec,", crbllrec=",crbllrec
-#ifndef STF
-    write(93,"(a)") "SIXTRACR> CRCHECK crbinrecs:"
-    do j=1,(crnapxo+1)/2
-      write(93,"(2(a,i0))") "SIXTRACR> ",j,": ",crbinrecs(j)
-    end do
-#endif
-    flush(93)
+    write(crlog,"(a)") "CR_CHECK> File "//cr_pntFile(nPoint)//" successfully read"
+    flush(crlog)
+    cr_pntRead(nPoint) = .true.
+    goto 100
```

Comment:
Can't this be replaced with a local flag which is flipped when checkpointing has sucessfully finished (followed by an `exit` to break the loop), and that flag checked and if it has not been sucessfully finished, print the error message below the loop and perror?

Reply:
Since there was already a flag for this purpose for the file exist logic, I reused it for file check pass here.